l10n_be_iso20022_pain/views/account_journal.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2024 ACSONE SA/NV
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record model="ir.ui.view" id="account_journal_form_view">
+        <field name="model">account.journal</field>
+        <field name="inherit_id" ref="account.view_account_journal_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//field[@name='invoice_reference_type']/parent::group"
+                position="attributes"
+            >
+                <attribute
+                    name="attrs"
+                >{'invisible': [('type', 'not in', ('sale', 'bank'))]}</attribute>
```

Comment:
I'm not sure it makes sense to make invoice_reference_model and invoice_reference_type visible for other journals than customer invoice journals. Because when we pay invoice bills with a journal, we don't control the type of reference we need to use for the suppliers. So I think invoice_reference_model and invoice_reference_type should only be used when generating payment reference on documents that we generate, not documents that we receive from suppliers.

Reply:
In this case, we remove the check on the payment reference in payment orders. We assume that the reference was checked for our invoices, and we don't concern ourselves with whether it's correct for vendor bills.